### PR TITLE
Modernize scripting and add support for running in a user session

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Retains the same behaviour on Core installs.

On classic the daemon defaults to doing nothing.

Enables running in a user session. (Need to manually connect wayland interface.)

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -7,16 +7,19 @@ grade: devel
...
-      - opengl
-      - wayland
+    restart-condition: on-failure
+
+  mir-kiosk-apps:
+    command: bin/mir-kiosk-apps
+plugs:
+  network:
+  opengl:
+  wayland:
```
Comment (gerboland) [inline snap/snapcraft.yaml:21]: Shouldn't these be 
```
 - network
 - opengl
 - wayland
```

**Review by gerboland — COMMENTED** [review]
One question, am in process of building & testing...

File: snap/snapcraft.yaml (hunk 1 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml:21]: No. This really is the format in a top level `plugs`, you're confusing it with `apps.<name>.plugs` which is still the way you write.

Comment (gerboland) [reply snap/snapcraft.yaml:21]: Indeed, it is a dict here. Ref: https://snapcraft.io/docs/snapcraft-yaml-reference

**Review by gerboland — APPROVED** [review]
Works well

**Pull request merged**